sunpy/io/_file_tools.py

```diff
@@ -84,7 +110,8 @@ def _read(filepath, function_name, filetype=None, **kwargs):
     pairs : `list`
         A list of (data, header) tuples.
     """
-    filepath = str(filepath)
+    filepath = get_open_file(filepath, **kwargs)
+    filepath = str(filepath.path)
```

Comment:
Why does it need to be cast to string? Is there other places in the code which break without this?